tests/unit/wer_report_test_wrapper.cpp

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#define _wer_report _wer_report_test
+#define AddVectoredExceptionHandler AddVectoredExceptionHandler_test
+#define SetThreadStackGuarantee SetThreadStackGuarantee_test
+#define RemoveVectoredExceptionHandler RemoveVectoredExceptionHandler_test
+#define WerReportCreate WerReportCreate_test
+#define WerReportAddDump WerReportAddDump_test
+#define WerReportSubmit WerReportSubmit_test
+#define WINBASEAPI
+
+// Included first to ensure the overridden copy is the only one.
+#include "wer_report.hpp"
+
+#undef _wer_report
+#undef AddVectoredExceptionHandler
+#undef SetThreadStackGuarantee
+#undef RemoveVectoredExceptionHandler
+#undef WerReportCreate
+#undef WerReportAddDump
+#undef WerReportSubmit
+#undef WINBASEAPI
+#undef _dupenv_s
+
+#include "catch_wrapper.hpp"
+
+uint32_t _vectored_exception_handler_test = 1;
+
+ULONG AddVectoredExceptionHandler_test_first = 0;
+PVECTORED_EXCEPTION_HANDLER AddVectoredExceptionHandler_test_handler = nullptr;
+
+_Ret_maybenull_ PVOID
+AddVectoredExceptionHandler_test(_In_ ULONG first, _In_ PVECTORED_EXCEPTION_HANDLER handler)
+{
+    AddVectoredExceptionHandler_test_first = first;
+    AddVectoredExceptionHandler_test_handler = handler;
+    return &_vectored_exception_handler_test;
+}
+
+unsigned long SetThreadStackGuarantee_test_stack_size_in_bytes = 0;
+
+BOOL
+SetThreadStackGuarantee_test(_Inout_ unsigned long* stack_size_in_bytes)
+{
+    SetThreadStackGuarantee_test_stack_size_in_bytes = *stack_size_in_bytes;
+    return TRUE;
+}
+
+ULONG
+WINAPI
+RemoveVectoredExceptionHandler_test(_In_ void* handle)
+{
+    UNREFERENCED_PARAMETER(handle);
+    return 0;
+}
+
+uint32_t WerReportCreate_test_report_handle = 2;
+
+bool WerReportCreate_test_fail = false;
+bool WerReportCreate_test_called = false;
+
+HRESULT
+WINAPI
+WerReportCreate_test(
+    _In_ const wchar_t* event_type,
+    _In_ WER_REPORT_TYPE report_type,
```

Comment:
```suggestion
    WER_REPORT_TYPE report_type,
```
Or does this have to conform to a (bad) prototype in a windows header file?

Reply:
(unresolving since I saw no change or answer to the question yet)

Reply:
My bad. Should have mentioned that this has to match the incorrect SAL annotation in the Windows header.